tests/accessor/accessor_default_values.h

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  This file provides functions for tests on accessor default values.
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_ACCESSOR_DEFAULT_VALUES_H
+#define SYCL_CTS_ACCESSOR_DEFAULT_VALUES_H
+
+#include "accessor_common.h"
+
+#include <type_traits>
+
+namespace accessor_default_values_test {
+
+namespace detail {
+constexpr int expected_dims = 1;
+constexpr sycl::target expected_target = sycl::target::device;
+constexpr sycl::access::placeholder expected_placeholder =
+    sycl::access::placeholder::false_t;
+template <typename T>
+constexpr auto expected_mode =
+    std::is_const_v<T> ? sycl::access_mode::read : sycl::access_mode::read_write
+}  // namespace detail
+
+/**
+ * @brief Provides functor that lets verify that local_accessor's template
+ *        parameters have expected default values
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ */
+template <typename T>
+class test_for_local_acc {
+ public:
+  void operator()(const std::string& type_name) {
+    auto section_name =
+        "Verify default value for dimensions template parameter for " +
+        type_name + " data type. [local_accessor]";
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::local_accessor<T>,
+                             sycl::local_accessor<T, detail::expected_dims>>);
+    }
+  }
+};
+
+/**
+ * @brief Provides functor that lets verify that host_accessor's template
+ *        parameters have expected default values
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ */
+
+template <typename T, typename AccessModeT, typename DimensionT,
+          typename TargetT>
+class test_for_host_acc {
+  static constexpr sycl::access_mode AccessMode = AccessModeT::value;
+  static constexpr int Dimension = DimensionT::value;
+  static constexpr sycl::target Target = TargetT::value;
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& access_mode_name,
+                  const std::string& target_name) {
+    auto section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for dimensions "
+                                    "template parameter. [host_accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::host_accessor<T>,
+                             sycl::host_accessor<T, detail::expected_dims>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for accessMode "
+                                    "template parameter. [host_accessor]");
+    SECTION(section_name) {
```

Comment:
It will give a same section name for different `Dimension` values.
See [Catch2 limitations]( https://github.com/catchorg/Catch2/blob/devel/docs/limitations.md#sections-nested-in-loops) for details. Also please be aware that you might have section names prototypes using https://github.com/KhronosGroup/SYCL-CTS/pull/337

Reply:
no, we will have the different section names with different `Dimension` values. The `Dimension` will be used [here](https://github.com/KhronosGroup/SYCL-CTS/pull/337/files#diff-8067235ffc00be4a68e8da8448acbf789b606099c5dbc1e969df066646bd2f64R154)